src/components/ChatEntry.js

```diff
@@ -1,22 +1,30 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
-const ChatEntry = (props) => {
+const ChatEntry = ({id, sender, body, timeStamp, liked, setLikes }) => {
+  const buttonClass = liked ?  '❤️':'🤍'
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name"> { sender } </h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p> { body } </p>
+        <TimeStamp time = {timeStamp} className="entry-time"></TimeStamp>
+        <button className="like" onClick = { () => setLikes(id)}>{buttonClass}
```

Comment:
This is an optimal approach to update the state of the heart without needing the entire entry, all you need to do is send the id of the liked entry up to App.js.